dna-src/minersweeper/index.ts

```diff
@@ -68,38 +68,40 @@ function getState(payload: {gameHash: Hash}): Action[] {
   return actions;
 }
 
-// function updateIdentity(payload: {newID: string}): boolean {
-//   try {
-//     debug("updating identity to: "+ payload.newID);
-//     updateAgent({Identity: payload.newID, Revocation: "x"});
-//     let result = update("agentID", App.Agent.String, App.Key.Hash);
-//     let updatedID = get(App.Key.Hash);
-//     return true;
-//   } catch (err) {
-//     return false
-//   }
-// }
-
-// function getIdentity(payload: {agentHash: Hash}): Hash | undefined {
-//   try {
-//     let h = get(payload.agentHash);
-//     return h;
-//   } catch (err) {
-//     return undefined;
-//   }
-// }
-
-// // function for batch getting a bunch of identities
-// function getIdentities(payload: {agentHashes: Hash[]}): [Hash, string][] {
-//   const result: [Hash, string][] = [];
-//   payload.agentHashes.forEach(hash => {
-//     const identity = getIdentity({agentHash: hash});
-//     if (identity !== undefined) {
-//       result.push([hash, identity]);
-//     }
-//   });
-//   return result;
-// }
+function updateIdentity(payload: {newID: string}): boolean {
+  try {
+    debug("updating identity to: "+ payload.newID);
+    let agentIdHash = getLinks(App.Key.Hash, "")[0].Hash;
+    debug(agentIdHash);
+    let result = update("agentID", payload.newID, agentIdHash);
+    return true;
+  } catch (err) {
+    debug(err);
+    return false
+  }
+}
+
+function getIdentity(payload: {agentHash: Hash}): string | undefined {
+  try {
+    let h = getLinks(payload.agentHash, "", {Load: true});
+    return h[0].Entry;
```

Comment:
Does this mean someone can never change their identity? Do we want that or do we want to take the last one?